### DISCUSSION TITLE
Add possibility to provide Homekit Data Points to KNX Bus

### DISCUSSION THREAD

**Discussion by cptbalusgit** [brutella/hkknx-public#319] [Ideas]
Idea: **Add a possibility to send HomeKit only data points to KNX Bus as GA**

Is it possible to send HomeKit Only data points e.g. humidity values from Bosch Devices which are integrated in HomeKit as HomeKit certified devices to the KNX bus via GA, so I consume this and use it on MDT Glass Push Button II?

I think a real HomeKit KNX Gateway should work in both ways ideally. With upcoming new devices with Matter communication standard it would allow real flexibility with the best from both worlds.

Obviously I am aware there are pure KNX possibilities for it, but we really all like the hkknx GW because it allows us to use Apple HomeKit.

**Comment by mbrockeu** [comment]
I have created a script in HKKNX and use the automations with shortcuts to call the url with the specific value. These are provided to the KNX bus. 

At the moment I use this for status of an Meross LED stripe which is directly connected to HomeKit.

**Reply by brutella** [reply]
Do I understand you correctly that you've created a local HTTP server which receives HTTP request from your iOS shortcut containing the value of a HomeKit device. You then parse the HTTP request payload and send that data to KNX, right?

**Reply by mbrockeu** [reply]
That's the code of the "API", not very nice but works.

```
var hkknx, fmt = import("hkknx"), import("fmt")
var http = import("net/http")

func startScene(w, r) {
	q = r.URL.Query()
	println(q)
	if q != nil {
		ga = r.URL.Query().Get("ga")
		value = r.URL.Query().Get("value")
		println(ga)
		println(value)
		if value == "Nein" || value == "No" {
			value = 0
		} else if value == "Ja" || value == "Yes" {
			value = 1
		}
		
		println(value)
		if value == 1 {
		hkknx.GroupWrite(true, ga)
		} else if value == 0 {
		hkknx.GroupWrite(false, ga)
		} else {
		hkknx.GroupWrite(value, ga)
		}
		//hkknx.SendNotification("Szene starten", fmt.Sprintf("Szene mit der GA: %s und Wert %s gestartet", ga, value))
	}
}

mux = http.NewServeMux()
mux.HandleFunc("/scene/", startScene)

http.ListenAndServe(":8082", mux)
```

Here the call in Home-App:

![IMG_3353](https://github.com/brutella/hkknx-public/assets/16849873/a3d82163-ff80-4dcd-a97d-deb005e2ab91)

I also have extended the call condition in Home+ to on/off

**Reply by cptbalusgit** [reply]
Interesting idea and kudos for the workaround. Not sure if it would scale very well for a lot more data points in a single setup/home. Especially if you are e.g. after humidity in % which might communicate all changes > 0,5% . Not sure how stable shortcuts and this approach would work for numerous HK only devices with higher communication frequency towards the knx bus. 
@brutella Can you see the benefit of the idea for the hkknx project?

**Comment by Pensive** [comment]
Just jumping in to +1 this. 

I've tried ABB VoiceControl, and the 1home server so far (unfortunately this is matter only) - I need to be able to Lock, and unlock - and receive statuses from - multiple Yale Assure Homekit Smart Locks, so you can unlock and lock it from knx switches.

Sadly every Homekit implementation I've found only goes one way......

I want to be able to discover homekit devices and use their status and control them from/on the knx bus. 

1home server would do this (with it's imminent update) but it is Matter only - and this is not a Matter lock.

**Reply by brutella** [reply]
The reason why most implementations only go "one way" is that once you pair an HomeKit accessory with a client (iOS) you cannot pair it again, for example with hkknx. This also means that if hkknx lets you pair a HomeKit accessory, you won't be able to pair it with the Apple Home app anymore.

I'm not sure if anyone would actually do that.

**Reply by Pensive** [reply]
Found a solution using Home Assistant - which works in this instance. Home Assistant directly connects via the Homekit Device integration - and acts as a homekit server to each Yale lock and I've been able to achieve this. Sadly - you are correct - this prevents the locks from beiong paired with a Homehub for the property. In this case - it doesn't matter. KNX is providing the UI system.

Long term - 1home Server will achieve this via matter integration but only with Matter locks.
